GetIntoTeachingApi/Services/CrmService.cs

```diff
@@ -45,6 +45,18 @@ public async Task<PrivacyPolicy> GetLatestPrivacyPolicy()
                 .First();
         }
 
+        public async Task<Candidate> GetCandidate(string email)
+        {
+            return (await _context.CreateQuery(ConnectionString(), "contact"))
+                .Where((contact) =>

```

Comment:
Can I just clarify - for this piece of code `.where` will generate the appropriate URL params for a query in dynamics, than `.select` is doing a client side 'reduce' to extract the first matching result?

Assuming the above is correct - two things I think are needed, haven't worked out if they are already there or not.
1. Some limit on the number of results returned
2. Need to retrieve the results by most-recent first

NB: Its out of scope of this piece of work but in SE we ended up tracking contacts we had matched before so we could favour those over new contacts added for the same email addresses.

Reply:
`Select` is a deferred operator; it will be executed once the results are loaded into memory by an immediate operator (`First` in this case). `First` also essentially equates to `Limit(1)` but returns a single value instead of a collection (I've maybe confused things by putting `Select` before `First`, but that seems to be the convention).

That's a good point about taking the most recent results; I had assumed that there would only be one contact with a particular email address in Dynamics. I'll add an ordering in by the created date 👍 we can maybe look at the last point at the end as a nice to have?

Reply:
Conclusion from discussion was the logic of this is still being finalised so we'll go with this for now and iterate it later if needed once the matching rules have been clarified.